packages/semcom-sdk/lib/component/services/register-component.service.ts

```diff
@@ -1,18 +1,38 @@
 import { AbstractRegisterComponentService, ComponentMetadata } from '@digita-ai/semcom-core';
 
 export class RegisterComponentService extends AbstractRegisterComponentService {
-  private registered: Map<string, string>;
+  private registered: Map<string, string> = new Map();
 
-  public async isRegistered(component: ComponentMetadata): Promise<boolean> {
-    if (!component) {
-      throw new Error('Argument component should be set.');
+  public async isRegistered(componentMetadata: ComponentMetadata): Promise<boolean> {
+    if (!componentMetadata || !componentMetadata.uri) {
+      throw Error('Invalid componentMetadata');
     }
 
-    return this.registered.has(component.uri);
+    return this.registered.has(componentMetadata.uri);
   }
 
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  public register(component: ComponentMetadata): Promise<ComponentMetadata> {
-    return;
+  public async register(componentMetadata: ComponentMetadata): Promise<string> {
+    if (!componentMetadata || !componentMetadata.tag || !componentMetadata.uri) {
+      throw Error('Invalid componentMetadata');
+    }
+
+    const isRegistered = await this.isRegistered(componentMetadata);
+
+    const component = await import(componentMetadata.uri);
+
+    if (isRegistered) {
+      const encoder = new TextEncoder();
+      const data = encoder.encode(componentMetadata.uri);
+      const hash = await crypto.subtle.digest('SHA-256', data);
+
+      componentMetadata = {
+        ...componentMetadata,
+        tag: `${componentMetadata.tag}-${hash}`,
+      };
```

Comment:
Because `componentMetadata` is a pointer, you're actually changing the component's tag outside of this function as well. Is this intended? If not, it might be best to assign the newly generated tag to a new variable.